### PR TITLE
refactor(ASFF): create class

### PR DESCRIPTION
**Pull request by jfagoagas**
### Context

We are creating modular outputs for Prowler, now is the time for the AWS Security Finding Format.

### Description

- Create the `ASFF(Output)` class.
- Add documentation for all the functions.
- Refactor the `filter_security_hub_findings_per_region` since now we don't need to recreate the ASFF findings, we've been doing it twice 🤦 
- Adapt the tests.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/outputs/csv/models.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def batch_write_data_to_file(self) -> None:
                 csv_writer.writeheader()
                 for finding in self._data:
                     csv_writer.writerow(finding)
+                self._file_descriptor.close()
```
Comment (jfagoagas) [inline prowler/lib/outputs/csv/models.py:42]: Added to the CSV class since it was missing.

File: prowler/lib/outputs/csv/models.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def transform(self, findings: list[Finding]) -> None:
     def batch_write_data_to_file(self) -> None:
         """Writes the findings to a file using the CSV format using the `Output._file_descriptor`."""
         try:
-            if self._file_descriptor:
+            if self._file_descriptor and not self._file_descriptor.closed:
```
Comment (jfagoagas) [inline prowler/lib/outputs/csv/models.py:33]: Good to check that the file descriptor is not closed.

File: prowler/lib/outputs/json_asff/json_asff.py
Comment (jfagoagas) [inline prowler/lib/outputs/json_asff/json_asff.py:1]: Move to the the `asff.py`

File: prowler/lib/outputs/json_asff/models.py
Comment (jfagoagas) [inline prowler/lib/outputs/json_asff/models.py:1]: Move to the the `asff.py`

File: prowler/lib/outputs/file_descriptors.py [hunk 3]
```diff
@@ -77,6 +76,8 @@ def fill_file_descriptors(output_modes, output_directory, output_filename, provi
                 # FIXME: Remove this once we always use the new CSV(Output)
                 if output_mode == "csv":
                     continue
+                elif output_mode == "json-asff":
+                    continue
```
Comment (MrCloudSec) [inline prowler/lib/outputs/file_descriptors.py:73]: Is this needed? As compliance does not use it, I think we can remove it

**Comment by MrCloudSec** [comment]
Tested locally and it perfectly works!

File: prowler/lib/outputs/file_descriptors.py (hunk 3 above)
Comment (jfagoagas) [reply prowler/lib/outputs/file_descriptors.py:73]: I think is needed for now until we solve Compliance, am I right @pedrooot ?

Comment (pedrooot) [reply prowler/lib/outputs/file_descriptors.py:73]: when all the output PRs are merged we can delete all the ifs and maintain only the specifics for compliance.

**Comment by pedrooot** [comment]
Now, `batch_send_to_security_hub` use `_send_findings_to_security_hub` passing the findings as a dict and the finding format in `AWSSecurityFindingFormat`. The real problem it's that `resolve_security_hub_previous_findings` pass the findings in dict function. I've made a "parse" in `_send_findings_to_security_hub` and depending the data structure it makes a different behaviour. This is not correct so I'll work on standardize all

**Comment by pedrooot** [comment]
The final approach it's that all the findings passed to `_send_findings_to_security_hub` are dict

File: prowler/lib/outputs/file_descriptors.py (hunk 3 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/file_descriptors.py:73]: I still think that this can be removed, this `output_mode` is not used in compliance...

**Review by pedrooot — APPROVED** [review]
giphy

**Pull request merged**